src/legacy/core_plugins/kibana/public/visualize/np_ready/editor/editor.js

```diff
@@ -372,22 +389,22 @@ function VisualizeAppController(
     $scope.timeRange = timefilter.getTime();
     $scope.opts = _.pick($scope, 'savedVis', 'isAddToDashMode');
 
-    stateMonitor = stateMonitorFactory.create($state, stateDefaults);
-    stateMonitor.ignoreProps(['vis.listeners']).onChange(status => {
-      $appStatus.dirty = status.dirty || !savedVis.id;
-    });
```

Comment:
Since you've removed the stateMonitor (which is good), I might be wrong but `$appStatus.dirty` is just initialized and no longer changed?

Reply:
Hey, yeah!
I a left one initializing reference to it intentionally!
I would highlight the thing about the `$appStatus`, because I didn't find a proper case how should it work!
So, if someone has a vision about how should this work, he can mention it here.
If it's redundant, I'll remove it!

Reply:
I'll take a closer look here

Reply:
`$appStatus.isDirty` is used when you want to share a visualization, but did has not been saved before, or there were changes after saving. 

https://github.com/elastic/kibana/blob/483ccc883f95b0e7db1772ee1df75b83c2549590/src/legacy/core_plugins/kibana/public/visualize/np_ready/editor/editor.js#L211

https://github.com/elastic/kibana/blob/483ccc883f95b0e7db1772ee1df75b83c2549590/src/legacy/core_plugins/kibana/public/visualize/np_ready/editor/editor.js#L222

When you then try to generate a report you get the following message: 

"Please save your work before generating a report."

So `$appStatus.isDirty` should be set to `true`, when the visualization was not saved, or it was saved and the `appState` changed. 
It should be set to `false` once the visualization was saved. And this persisted state is the state that you need to check agains for further changes that modify  `$appStatus.isDirty`

Reply:
@kertal thanks for the deep explanation!
Applied dirty state checker due to current code